apps/condo/domains/acquiring/schema/CreatePaymentByLinkService.js

```diff
@@ -67,15 +68,45 @@ const CreatePaymentByLinkService = new GQLCustomSchema('CreatePaymentByLinkServi
                     paymPeriod, // mm.yyyy
                     sum,
                     persAcc, // resident's account within organization
-                } = getQRCodeFields(qrCodeFields, ['personalAcc', 'paymPeriod', 'sum', 'persAcc'])
+                    payeeINN,
+                } = getQRCodeFields(qrCodeFields, ['personalAcc', 'paymPeriod', 'sum', 'persAcc', 'payeeINN'])
                 const period = formatPeriodFromQRCode(paymPeriod)
                 const amount = String(Big(sum).div(100))
 
-                const auxiliaryData = await findAuxiliaryData(qrCodeFields, { address: ERRORS.ADDRESS_IS_INVALID })
-                const { normalizedAddress } = auxiliaryData
+                // Existence of this billing account was checked at ValidateQRCodeService
+                const billingAccounts = await find('BillingAccount', {
+                    number: persAcc,
+                    context: {
+                        organization: { tin: payeeINN, deletedAt: null },
+                        status: BILLING_CONTEXT_FINISHED_STATUS,
+                        deletedAt: null,
+                    },
+                    deletedAt: null,
+                })
+
+                const [billingAccount] = billingAccounts
```

Comment:
You can merge this 2 calls into one 
const [billingAccount] = await find('BillingAccount', {
and you need to add a check if no account found